### PR TITLE
Microsoft teams 1839 - CI tests failing

### LINKED ISSUE #1839: Microsoft Teams source connector flaky tests
## Expected Behavior
I expect the Microsoft teams source connector CI to consistently succeed. 

## Current Behavior
Seems to be flaky, ~60% success rate: https://github.com/airbytehq/airbyte/runs/1766246384

## Acceptance Criteria
1. Understand where the flakiness is coming from and implement a fix for it

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## How
Installing backoff - sometime Microsoft Graph API returns empty response for channels response. It caused error for this stream and related. Using backoff helps to handle this case.

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]

**Comment by sherifnada** [comment]
@vitaliizazmic build is failing -- can you take a look? It seems like it comes from line 105 in `client.py`

```
value = raw_response.get("value", [])
            if not value:
                raise requests.exceptions.RequestException()
```

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py [hunk 1]
```diff
@@ -102,8 +106,6 @@ def _make_request(self, api_url: str, params: Optional[Dict] = None) -> Union[Di
 
     @staticmethod
     def _get_response_value_unsafe(raw_response: Dict) -> List:
-        if "value" not in raw_response:
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py:105]: why was this removed? your description in the PR made it sound like this field might not always be populated.

**Review by sherifnada — COMMENTED** [review]
@vitaliizazmic I don't entirely understand the problem being solved. Is the MSFT API randomly returning empty responses? Do you know under what conditions it returns an empty response and under what condition it returns a non-empty response? I want to move forward here but I don't have a good understanding of what exactly is the problem, when it occurs, and how this change fixes it.

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py [hunk 2]
```diff
@@ -83,6 +84,9 @@ def _get_access_token(self) -> str:
         else:
             raise MsalServiceError(error=result.get("error"), error_description=result.get("error_description"))
 
+    @backoff.on_exception(
+        backoff.expo, (requests.exceptions.ConnectionError, MsalServiceError, requests.exceptions.RequestException), max_tries=7
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py:88]: could you add a log message which says we are backing off? e.g: `Encountered exception when querying the Microsoft API: <exception>. Backing off`.

**Comment by vitaliizazmic** [comment]
> @vitaliizazmic I don't entirely understand the problem being solved. Is the MSFT API randomly returning empty responses? Do you know under what conditions it returns an empty response and under what condition it returns a non-empty response? I want to move forward here but I don't have a good understanding of what exactly is the problem, when it occurs, and how this change fixes it.

@sherifnada channels related streams require channel id. To get channel ids I use [list channels](https://docs.microsoft.com/en-us/graph/api/channel-list?view=graph-rest-1.0&tabs=http) method. To decrease response size I used $select OData query parameters and set it value to "id".  It should returns only ids for channels. But empty value have been received lately. When API is used without query parameter, it returns channel. Therefore I deleted this parameter. There are several open issues connected wit this problem:
- https://github.com/microsoftgraph/microsoft-graph-docs/issues/11494
- https://github.com/microsoftgraph/microsoft-graph-docs/issues/11472

File: airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py [hunk 3]
```diff
@@ -173,8 +184,7 @@ def get_channels(self):
 
     def _get_channel_ids(self, group_id: str):
         api_url = self._get_api_url(f"teams/{group_id}/channels")
-        params = {"$select": "id"}
-        channels_ids = self._get_response_value_unsafe(self._make_request(api_url, params=params))
+        channels_ids = self._get_response_value_unsafe(self._make_request(api_url))
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-microsoft-teams/source_microsoft_teams/client.py:187]:
```suggestion
        # TODO: pass params={"$select": "id"} to make_request once the related bug in the MSFT API
        # is fixed: microsoftgraph/microsoft-graph-docs#11494
        channels_ids = self._get_response_value_unsafe(self._make_request(api_url))
```

**Review by sherifnada — APPROVED** [review]
Welp, looks like a legit error on Microsoft's part. Nicely done @vitaliizazmic 🎉
